src/string_search.h

```diff
@@ -8,6 +8,15 @@
 #include "node.h"
 #include <string.h>
 
+#ifdef DEBUG_STRING_SEARCH
+#include <stdio.h>
+#define DEBUG_TRACE(s) printf("%s search %s\n", \
+    subject.forward() ? "forward" : "reverse", s);
+#else
+#define DEBUG_TRACE(s)  // no-op
```

Comment:
Removed. Note though that this was the only way I noticed some serious missing test coverage: the whole Boyer-Moore algorithm (nontrivial code, rarely exercised) was never reached by the unit tests before. (The unit tests previously only cover Boyer-Moore-Horspool, Linear, and Single-Char. After this PR they cover Boyer-Moore as well.)